src/Credentials/CredentialProvider.php

```diff
@@ -63,8 +68,9 @@ public static function defaultProvider(array $config = [])
     {
         return self::memoize(
             self::chain(
-                self::env(),
-                self::ini(),
+                self::cache($config),
```

Comment:
I wasn't saying the memoize function should have external caching built into it, but rather that there could be another function similar to the memoize function that allows for a customizable external cache. There may be some refactoring that could be done to share code, but I think these are separate abstractions.